### DISCUSSION TITLE
Substitution for differential equations

### DISCUSSION THREAD

**Discussion by Alexishul** [Mathics3/mathics-core#848] [Q&A]
Hello,

For a project dealing with differential equations, i would like to replace all the variables "y" of my differential equation with another expression, like the sympy.subs() method actually does.

However i want to get totally rid of Sympy and only use Mathics.
Is this possible ? I started to look into "Rule" and "Pattern" classes, but i didn't really understand...

Here's an example of what i expect : 
```
session = MathicsSession()

x = Symbol("x")
y = Symbol("y")

eq1str = "Derivative[y, {x, 2}] + 100 * y"
eq1 = session.evaluate(eq1str).to_python()

eq2str = "cos(x) + x**2"
eq2 = session.evaluate(eq2str).to_python()

#Here : in "eq1" replace "y" by "eq2
```

**Comment by mmatera** [comment]
Usually, in WMA, I make the replace like this:

```
eq1/.{y->(Cos[#1]+#1^2)&}
```

The reason to use an anonymous function is that in the equation,  `D[y[x],{x,2]]` is evaluated as `Derivative[2][y][x]`, so
```
eq1= y''[x]+y[x]==0
```
actually stores in `eq1`
```
eq1= Derivative[2][y][x]+y[x]==0
```
Then, to get the equation evaluated, `y` must be replaced by something that can be applied over the argument x:

```
(Cos[#1]+#1^2)&[x] === Cos[x]+x^2
```

**Comment by Alexishul** [comment]
So, if I understand well, on Python, this should be correct ?
```
eq1str = "Derivative[y, {x, 2}] + 100 * y"
eq1 = session.evaluate(eq1str).to_python()

eq2str = "5 * x"
eq2 = session.evaluate(eq2str).to_python()

#Here : in "eq1" replace "y" by "eq2

eq3str = eq1str + "/.{y->(" + eq2str + ")}"
eq3 = session.evaluate(eq3str).to_python()
print(eq3)
```

Here is the output : 
```
System`Plus[System`Times[500, Global`x], System`Derivative[System`Times[5, Global`x], {Global`x,2}]]`
```
which means : 
```
500x + (5x)''
```

Tell me if i'm wrong, and if not, how can i simplify this result ? 
(the simplification `(5x)'' = 0`)
